server/src/main/java/org/elasticsearch/search/profile/AbstractProfileBreakdown.java

```diff
@@ -25,34 +28,46 @@ public abstract class AbstractProfileBreakdown<T extends Enum<T>> {
     /**
      * The accumulated timings for this query node
      */
-    private final Timer[] timings;
+    private final Map<Integer, List<Timer>> timings;
     private final T[] timingTypes;
 
     /** Sole constructor. */
     public AbstractProfileBreakdown(Class<T> clazz) {
         this.timingTypes = clazz.getEnumConstants();
-        timings = new Timer[timingTypes.length];
-        for (int i = 0; i < timings.length; ++i) {
-            timings[i] = new Timer();
+        timings = new HashMap<>(this.timingTypes.length);
```

Comment:
use `Maps.mapWithExpectedSize(this.timingTypes.length)`. If you statically set the size like this, it doesn't respect the loadFactor and you will still get a hashmap size increase as you add these values.

Reply:
nice, didn't know about that detail.